teckyl/MLIRAffineExprGen.h

```diff
@@ -69,7 +69,7 @@ class MLIRAffineExprGen {
 
         if (uintval >
             static_cast<uint64_t>(std::numeric_limits<int64_t>::max()))
-          throw Exception("Unsigned integer constant too big");
+          llvm_unreachable("Unsigned integer constant too big");
```

Comment:
I think it would make sense to keep this exception. There is nothing fundamentally wrong with large integer constants and a caller might want to catch this case in order to provide more detailed feedback to the user. This could be wrapped in a macro that either evaluates to `llvm_unreachable` or that throws depending on a compiler flag.